libs/domain/checkout/billing-address/billing-address.component.ts

```diff
@@ -45,7 +45,7 @@ export class CheckoutBillingAddressComponent
     if (
       deliveryAddress &&
       this.$isSameAsShippingAddress() &&
-      !this.$addresses()?.length
+      this.$addresses()?.length === 0
```

Comment:
Is it important to check exact equality to 0 instead of just falsy check?

Reply:
yup, otherwise it returns false if the addresses are not yet loaded